### ISSUE TITLE
Consistent `new` signature between Ident and Lifetime

### ISSUE THREAD

**Issue by dtolnay** [dtolnay/syn#344]
Currently we have:

```rust
impl Ident {
    fn new(&str, Span) -> Self;
}

impl Lifetime {
    fn new(Term, Span) -> Self;
}
```

I don't know a good reason that we should keep one of these taking &str and the other Term. I don't know which one is better but I would prefer if they could be consistent.

**Comment by mystor** [comment]
I think that I'd pick `new(&str, Span)`, because I think that's more likely to be used, and have a `from_term` version for building them from `Term`s.

**Comment by clarcharr** [comment]
Why not `Into<Term>`?

**Comment by dtolnay** [comment]
Yes I would prefer `new(&str, Span)` and `From<Term>`.

**Comment by dtolnay** [comment]
Fixed in e2099f3b864a8fbdfcb06f503f3ffb6da0ecaed7.
